src/components/encoding-pane/index.tsx

```diff
@@ -43,15 +46,21 @@ class EncodingPanelBase extends React.PureComponent<EncodingPanelProps, {}> {
 
   public render() {
     const {specPreview, spec} = this.props;
+    const fieldCount = this.props.schema.fieldSchemas.length;
+    const {manualSpecificationOnly} = this.props.config;
     const {anyEncodings} = specPreview || spec;
 
     const positionShelves = ['x', 'y'].map(this.encodingShelf, this);
     const facetShelves = ['row', 'column'].map(this.encodingShelf, this);
     const nonPositionShelves = ['size', 'color', 'shape', 'detail', 'text'].map(this.encodingShelf, this);
-    const wildcardShelves = [
-      ...anyEncodings.map((_, i) => i),
-      -1 // map the empty placeholder to -1
-    ].map(this.wildcardShelf, this);
+    const wildcardShelvesGroup = !manualSpecificationOnly && fieldCount > 0 ? (
```

Comment:
I don't understand why do you have to check `fieldCount` here. 

Plus, why is it related to a PR for `manualSpecificationOnly`?